core/src/main/java/com/yandex/yoctodb/v1/mutable/segment/V1StoredIndex.java

```diff
@@ -78,7 +80,9 @@ public OutputStreamWritable buildWritable() {
                 padded.add(empty);
                 expectedDocument++;
             }
-
+            if (padded.contains(e.getValue())) {
```

Comment:
This gives you a quadratic complexity for just checking uniqueness. Consider using `LinkedHashSet`s `add` method output instead of `ArrayList`.
Anyway, I think it's better to check uniqueness on index construction and not on serialization time.